library/spdm_responder_lib/libspdm_rsp_set_certificate.c

```diff
@@ -194,7 +194,9 @@ libspdm_return_t libspdm_get_response_set_certificate(libspdm_context_t *spdm_co
 
     /* set certificate to NV*/
     result = libspdm_write_certificate_to_nvm(slot_id, cert_chain,
-                                              cert_chain_size);
+                                              cert_chain_size,
+                                              spdm_context->local_context.algorithm.base_asym_algo,
```

Comment:
`spdm_context->connection_info.algorithm.base_asym_algo`, else there might be multiple algorithms in the bitmask.